Assets/RedBlueGames/BulkRename/Editor/BulkRenamerWindow.cs

```diff
@@ -23,712 +23,755 @@
+		{
+			gameObject.name = newName;
+		}
+
+		public 	static string ReplaceSpritePrefixInMetafile (string metafileText, string prefixToReplace, string newPrefix)
+		{			
+			string fileIDPattern = prefixToReplace;
+			var fileIDRegex = new System.Text.RegularExpressions.Regex (fileIDPattern);
+			string replacementText = newPrefix;
+			return fileIDRegex.Replace (metafileText, replacementText);
+		}
+
+		void RenameSprites (Dictionary<Object,string> allSpritesWithNewNames)
+		{
+			string actualAssetName;
+			string newAssetName;
+			string path;
+			HashSet<string> assetsToReimport = new HashSet<string> ();
+
+			foreach (var asset in allSpritesWithNewNames) {
+				actualAssetName = asset.Key.name;
+				newAssetName = asset.Value;
+				path = AssetDatabase.GetAssetPath (asset.Key);
+				TextureImporter parentAsset = TextureImporter.GetAtPath (path)as TextureImporter;
+				#region Changing name in metafiles
+				PropertyInfo cachedInspectorModeInfo = typeof(SerializedObject).GetProperty ("inspectorMode", BindingFlags.NonPublic | BindingFlags.Instance);
+				SerializedObject so = new UnityEditor.SerializedObject (asset.Key);
+				cachedInspectorModeInfo.SetValue (so, InspectorMode.Debug, null);
+				PropertyInfo inspectorModeInfo = typeof(SerializedObject).GetProperty ("inspectorMode", BindingFlags.NonPublic | BindingFlags.Instance);
+				SerializedProperty sp = so.FindProperty ("m_LocalIdentfierInFile");
+				string oldNameMeta = sp.intValue + ": " + actualAssetName;
+				string newNameMeta = sp.intValue + ": " + asset.Value;
+				string metaFile = System.IO.File.ReadAllText (path + ".meta");
+				string ammendedMetaFile = ReplaceSpritePrefixInMetafile (metaFile, actualAssetName, newAssetName);
+				System.IO.File.WriteAllText (path + ".meta", ammendedMetaFile);
+				#endregion
+			}
+			AssetDatabase.Refresh ();// it has local range - only refresh assets in body of this method!
+
+
+			#region Changing names in Project Window
+			foreach (var asset in allSpritesWithNewNames) {
+				actualAssetName = asset.Key.name;
+				newAssetName = asset.Value;
+				path = AssetDatabase.GetAssetPath (asset.Key);
+				TextureImporter parentAsset = TextureImporter.GetAtPath (path)as TextureImporter;
+				SpriteMetaData[] spriteMetaData = parentAsset.spritesheet;
+
+				for (int i = 0; i < spriteMetaData.Length; i++) {
+					if (spriteMetaData [i].name.Equals (actualAssetName)) {
+						spriteMetaData [i].name = newAssetName;
+						break;                   
+					}	
+				}
+				EditorUtility.SetDirty (parentAsset);                            
+				parentAsset.spritesheet = spriteMetaData;
+				assetsToReimport.Add (path);
+			}
+			foreach (var item in assetsToReimport) { 
+				AssetDatabase.ImportAsset (item, ImportAssetOptions.ForceUpdate);
+			}
+			#endregion
+		}
+
+		private void RenameAsset (UnityEngine.Object asset, string newName)
+		{
+			if (asset is Sprite) {
+				allSpritesWithNewNames.Add (asset, newName);
```

Comment:
This member variable, `allSpritesWithNewNames` is a side effect to this function. Sprites aren't renamed, but they are added to a list. I expect this function to rename the thing.

I *wish* we could do it this way, but I think for performance we need to think of something else.

I'm thinking the cleanest thing to do it is to separate the assets into three lists inside the `RenameAssets` function: Non-Sprite Assets, GameObjects, and Assets. Then have separate functions for renaming lists of each type. Think that makes sense?